pandasai/smart_dataframe/__init__.py

```diff
@@ -550,178 +180,55 @@ def last_error(self):
-
-    @property
-    def save_charts_path(self):
-        return self.lake.save_charts_path
-
-    @save_charts_path.setter
-    def save_charts_path(self, save_charts_path: str):
-        self.lake.save_charts_path = save_charts_path
-
-    @property
-    def custom_whitelisted_dependencies(self):
-        return self.lake.custom_whitelisted_dependencies
-
-    @custom_whitelisted_dependencies.setter
-    def custom_whitelisted_dependencies(
-        self, custom_whitelisted_dependencies: List[str]
-    ):
-        self.lake.custom_whitelisted_dependencies = custom_whitelisted_dependencies
-
-    @property
-    def max_retries(self):
-        return self.lake.max_retries
-
-    @max_retries.setter
-    def max_retries(self, max_retries: int):
-        self.lake.max_retries = max_retries
-
     @property
     def llm(self):
         return self.lake.llm
 
-    @llm.setter
-    def llm(self, llm: Union[LLM, LangchainLLM]):
-        self.lake.llm = llm
-
-    @property
-    def table_name(self):
-        return self._table_name
-
-    @property
-    def table_description(self):
-        return self._table_description
-
-    @property
-    def custom_head(self):
-        data = StringIO(self._custom_head)
-        return pd.read_csv(data)
-
-    @custom_head.setter
-    def custom_head(self, custom_head: pd.DataFrame):
-        self._custom_head = custom_head.to_csv(index=False)
-
     @property
     def last_query_log_id(self):
-        return self._lake.last_query_log_id
+        return self.lake.last_query_log_id
 
     def __getattr__(self, name):
-        if name in self._core.__dir__():
-            return getattr(self._core, name)
-        elif name in self.dataframe.__dir__():
-            return getattr(self.dataframe, name)
-        else:
-            return self.__getattribute__(name)
+        return self.dataframe_proxy.__getattribute__(name)
 
     def __getitem__(self, key):
-        return self.dataframe.__getitem__(key)
+        return self.dataframe_proxy.__getitem__(key)
 
     def __setitem__(self, key, value):
-        return self.dataframe.__setitem__(key, value)
+        return self.dataframe_proxy.__setitem__(key, value)
 
     def __dir__(self):
-        return dir(self._core) + dir(self.dataframe) + dir(self.__class__)
+        return dir(self.dataframe_proxy) + dir(self.__class__)
 
     def __repr__(self):
-        return self.dataframe.__repr__()
+        return self.dataframe_proxy.__repr__()
 
     def __len__(self):
-        return len(self.dataframe)
+        return self.dataframe_proxy.connector.rows_count
 
     def __eq__(self, other):
-        if isinstance(other, self.__class__) and (
-            self._core.has_connector and other._core.has_connector
-        ):
-            return self._core.connector.equals(other._core.connector)
-
-        return False
-
-    def is_connector(self):
-        return self._core.has_connector
+        return self.dataframe_proxy.connector.equals(other.dataframe_proxy.connector)
```

Comment:
@gventuri  .equals doens't exists for PandasConnector if gives error when using with direct_sql: true config.
"Unfortunately, I was not able to answer your question, because of the following error:\n\n'PandasConnector' object has no attribute 'equals'\n"